### PR TITLE
Fix style update with emojis on iOS

### PR DESCRIPTION
**Pull request by tomekzaw**
### Details

This PR fixes formatting of the Markdown string after `style` and `markdownStyle` of `MarkdownTextInput` are updated, in case when the value contains an emoji.

https://github.com/Expensify/react-native-live-markdown/assets/20516055/92c08748-9f75-475e-bff3-dee54d3d29c9

This was previously fixed in #269 but it turns out that it only works correctly if the string does not contain any emojis.

The root cause of this issue is that `setAttributedText` skips the update if the attributed string are, however, if the string contains an emoji, the fallback mode is used and only the raw strings are compared:

https://github.com/facebook/react-native/blob/58ddd74202bede2238768def01898741606ec64f/packages/react-native/Libraries/Text/TextInput/RCTBaseTextInputView.mm#L128-L165

Screenshot 2024-04-17 at 11 33 09

### Related Issues

Fixes https://github.com/Expensify/react-native-live-markdown/issues/293.

### Manual Tests

Repro: https://github.com/Expensify/react-native-live-markdown/issues/293#issuecomment-2060096528

### CODE REVIEW

**Review by Skalakid — COMMENTED** [review]
LGTM however GH actions shows some warnings about undeclared selectors

**Comment by tomekzaw** [comment]
TODO: check in singleline mode

File: ios/MarkdownTextInputDecoratorView.mm [hunk 1]
```diff
@@ -100,11 +100,13 @@ - (void)setMarkdownStyle:(RCTMarkdownStyle *)markdownStyle
...
   [_markdownUtils setMarkdownStyle:markdownStyle];
+
+  // apply new styles
 #ifdef RCT_NEW_ARCH_ENABLED
-  [_textInput textInputDidChange]; // apply new styles
+  [_textInput _setAttributedString:_textView.attributedText];
 #else
-  [_textInput setAttributedText:_textInput.attributedText]; // apply new styles
-#endif
+  [_textInput setAttributedText:_textView.attributedText];
```
[the remark this answers is not in the document]
Comment (tomekzaw) [reply ios/MarkdownTextInputDecoratorView.mm]: You're right (as always), changed in 8720e3e, now it works both for singleline and multiline.

**Review by j-piasecki — APPROVED** [review]
LGTM, please make CI happy 😄

**Comment by tomekzaw** [comment]
I forgot about Fabric 😢

**Review by j-piasecki — APPROVED** [review]

**Pull request merged**